src/core/surface/alarm.c

```diff
@@ -63,9 +63,9 @@ grpc_alarm *grpc_alarm_create(grpc_completion_queue *cq, gpr_timespec deadline,
   alarm->cq = cq;
   alarm->tag = tag;
 
+  grpc_cq_begin_op(cq, tag);
```

Comment:
We have to mark that the op is begun before we can actually kick it off by doing the timer_init. Otherwise the op could actually get marked done before we "begin" it (because it can get processed on another thread, since alarms stop other pluckers or nexters).